pkg/common/log/reopen_test.go

```diff
@@ -0,0 +1,75 @@
+package log
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReopenOnSignal(t *testing.T) {
+	const (
+		_tempDir         = "spirelogrotatetest"
+		_testLogFileName = "test.log"
+		_rotatedSuffix   = "rotated"
+		_firstMsg        = "a message"
+		_secondMsg       = "another message"
+	)
+	dir, err := os.MkdirTemp("", _tempDir)
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	logFileName := filepath.Join(dir, _testLogFileName)
+	rotatedLogFileName := logFileName + "." + _rotatedSuffix
+	rf, err := NewReopenableFile(logFileName)
+	require.NoError(t, err)
+
+	fsInfo, err := rf.f.Stat()
+	require.NoError(t, err)
+	assert.Equal(t, int64(0), fsInfo.Size(), "%s should be empty", fsInfo.Name())
+
+	logger, err := NewLogger(WithReopenableOutputFile(rf))
+	require.NoError(t, err)
+
+	logger.Warning(_firstMsg)
+
+	fsInfo, err = rf.f.Stat()
+	require.NoError(t, err)
+	initialLogSize := fsInfo.Size()
+	assert.NotEqual(t, int64(0), fsInfo.Size(), "%s should not be empty", fsInfo.Name())
+
+	signalCh := make(chan os.Signal, 1)
+	// prevent test from hanging until test runner timeout when failing
+	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond)
```

Comment:
We generally steer away from introducing test-specific timeouts. I would prefer if we remove this since we do not want to rely on any test timing other than the test passing in the timeframe allowed by `go test` for the package.

Reply:
Ack.